### ISSUE TITLE
How do we do ci-builds?

### ISSUE THREAD

**Issue by fir3pho3nixx** [castleproject/Windsor#395]
We have moved from TeamCity to AppVeyor. I was wondering if we should leave some documentation around for how to attain pre-release versions of Castle Windsor/Core with some build tips. This would help the community if they wanted to try the build artifacts of live PR's. 

I do know from [here](https://github.com/castleproject/Windsor/issues/393#issuecomment-380329049) we should probably agree that if appveyor is a thing which should at least put a small description with links in the README.md. 

Thinking light touch but with a huge impact. It does have the down side of tying us to appveyor/travis (in the future) though.

**Comment by jonorossi** [comment]
I don't mind that it would tie us to one of the CI platforms because we wouldn't guarantee any longevity for CI builds. If you are talking about proper named prerelease builds, we can publish them to nuget.org like we've done in the past.

If you add this to the readme we should probably merge it with the text we've already got in Castle Core about debugging symbols (https://github.com/castleproject/Core#releases), the debugging symbols will probably change in the future when nuget.org supports that better, but for now it'll be good have it consistent between both projects.

**Comment by jonorossi** [comment]
With respect to the title, I'd rename it to "CI builds" or something so indicate you didn't mean named pre-releases.

**Comment by fir3pho3nixx** [comment]
> I don't mind that it would tie us to one of the CI platforms because we wouldn't guarantee any longevity for CI builds. If you are talking about proper named prerelease builds, we can publish them to nuget.org like we've done in the past.

Super good. How do we doc this? You know give our users good steer.

**Comment by fir3pho3nixx** [comment]
@jonorossi ^^

Included a pre-release for master also. Not sure how appveyor expose this as a nuget feed though.

**Comment by fir3pho3nixx** [comment]
Can I PR?

**Comment by jonorossi** [comment]
As I mentioned above (quoted below) I don't think referring to every build as a prerelease build, as most people understand prerelease builds as `-alpha`, `-beta`, etc ones that are somewhat supported. Do you dislike the build status badge which links to AppVeyor?

> With respect to the title, I'd rename it to "CI builds" or something to indicate you didn't mean named pre-releases.

**Comment by fir3pho3nixx** [comment]
> Do you dislike the build status badge which links to AppVeyor?

Let me add it in.

**Comment by fir3pho3nixx** [comment]
>  Do you dislike the build status badge which links to AppVeyor?

It is pretty cool, if this is not there I will make sure it is. A few problems with appveyor. 

 - They are switching to a 6 month rolling cycle for storing artifacts. They are trying to cut down on their storage costs for OSS I think. 
- I briefly googled around for how you consume nugets from them but did not notice a tidy way of setting up a local appveyor feed. https://www.appveyor.com/docs/nuget/.

What do you think?

**Comment by jonorossi** [comment]
> They are switching to a 6 month rolling cycle for storing artifacts. They are trying to cut down on their storage costs for OSS I think.

They are doing it for both paid and free plans, however they had a recent follow up post that says that NuGet packages won't be deleted after 6 months: https://www.appveyor.com/blog/2018/06/05/artifacts-retention-policy-update/

**Comment by fir3pho3nixx** [comment]
That is good news. Although I am getting a sense of cost cutting angst which could bite at any time. 

> I briefly googled around for how you consume nugets from them but did not notice a tidy way of setting up a local appveyor feed. https://www.appveyor.com/docs/nuget/.

Do we want to automate anything in this space?

**Comment by jonorossi** [comment]
> That is good news. Although I am getting a sense of cost cutting angst which could bite at any time.

Definitely, the storage costs would be quite a lot, however as I've said before I don't think we really have to worry about it as we wouldn't guarantee any longevity for CI builds, anyone using the CI builds should be using the newest build otherwise they should use a released and supported version from nuget.org. So even if it did change to 6 months for NuGet project feeds (i.e. https://ci.appveyor.com/nuget/windsor-qkry8n2r6yak) then I don't think that would be a problem.

Symbols though are a different story, and I think the solution there is to wait until the NuGet team get whatever their symbol package solution done since they've taken over control of symbolsource.org.

**Comment by fir3pho3nixx** [comment]
> Symbols though are a different story, and I think the solution there is to wait until the NuGet team get whatever their symbol package solution done since they've taken over control of symbolsource.org.

Wow, did not know that. Sounds like it has the potential to be quite disruptive. :)

I think I might add a link to the README.md? Something like:

```
NuGet Preview Feed: https://ci.appveyor.com/nuget/windsor-qkry8n2r6yak
```

**Comment by fir3pho3nixx** [comment]
Closed via #435
